src/Routing/Route/Route.php

```diff
@@ -423,6 +423,7 @@ public function match(array $url, array $context = [])
             $hostOptions['_ext'] = $url['_ext'];
             unset($url['_ext']);
         }
+        unset($defaults['_ext']);
```

Comment:
It's not the only difference. 

According to documentation, passing `$options` to `Router::scope()`, should make those automatically applied to all routes created under that scope. However, when creating routes using `$routes->resources()` in that scope, the scope's options are not applied to the routes it creates but if you create it manually with `$routes->connect()` it does.

Here's a test case that explains it better probably:

```diff
diff --git a/tests/TestCase/Routing/RouterTest.php b/tests/TestCase/Routing/RouterTest.php
index c27837f..cc1e79b 100644
--- a/tests/TestCase/Routing/RouterTest.php
+++ b/tests/TestCase/Routing/RouterTest.php
@@ -1108,6 +1108,29 @@ class RouterTest extends TestCase
     }

     /**
+     * testRestUrlGenerationWithExtensions method
+     *
+     * @return void
+     */
+    public function testResourcesCreationWithCustomParams()
+    {
+        Router::scope('/', ['id' => '-?[\d]+'], function ($r) {
+            $r->extensions('json');
+            $r->resources('Articles');
+        });
+
+        $result = Router::url([
+            'plugin' => null,
+            'controller' => 'Articles',
+            'action' => 'edit',
+            'id' => '-1',
+            '_method' => 'GET',
+        ]);
+        $expected = '/articles/edit/-1';
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
      * test url() when the current request has an extension.
      *
      * @return void
```